server/src/server.ts

```diff
@@ -83,11 +88,15 @@ connection.onShutdown(() => {
   disposables.forEach((disposable) => { disposable.dispose() })
 })
 
-// eslint-disable-next-line @typescript-eslint/no-misused-promises
-connection.onDidChangeConfiguration(async (change) => {
+connection.onDidChangeConfiguration((change) => {
   logger.level = change.settings.bitbake.loggingLevel
   parseOnSave = change.settings.bitbake.parseOnSave
   eSDKMode = change.settings.bitbake.eSDKMode
+  if (workspaceFolder !== undefined) {
+    // eslint-disable-next-line no-template-curly-in-string
+    const bitbakeFolder = (change.settings.bitbake.pathToBitbakeFolder as string).replace('${workspaceFolder}', workspaceFolder)

```

Comment:
You should use the resolveSettingPath from client/src/lib/src/BitbakeSettings.ts to remove code duplication.

Reply:
done